engine/consensus/dkg/reactor_engine.go

```diff
@@ -273,20 +364,22 @@ func (e *ReactorEngine) registerPhaseTransition(view uint64, fromState dkgmodule
 func (e *ReactorEngine) end(epochCounter uint64) func() error {
 	return func() error {
 		err := e.controller.End()
-		if err != nil {
-			if crypto.IsDKGFailureError(err) {
-				e.log.Warn().Msgf("node %s with index %d failed DKG locally: %s", e.me.NodeID(), e.controller.GetIndex(), err.Error())
-			} else {
-				return err
+		if crypto.IsDKGFailureError(err) {
+			e.log.Warn().Msgf("node %s with index %d failed DKG locally: %s", e.me.NodeID(), e.controller.GetIndex(), err.Error())
+			err := e.dkgState.SetDKGEndState(epochCounter, flow.DKGEndStateDKGFailure)
+			if err != nil {
+				return fmt.Errorf("failed to set dkg end state following dkg end error: %w", err)
 			}
+		} else if err != nil {
+			return fmt.Errorf("unknown error ending the dkg: %w", err)
 		}
 
 		privateShare, _, _ := e.controller.GetArtifacts()
 
 		privKeyInfo := encodable.RandomBeaconPrivKey{
 			PrivateKey: privateShare,
 		}
```

Comment:
I think we don't have to wrap the key here as it's done already inside `InsertMyBeaconPrivateKey`

Reply:
https://github.com/onflow/flow-go/pull/1721/commits/5f62cb3b0f49c59b2118a4387a621b74a5516f03